src/library.js

```diff
@@ -1547,7 +1547,37 @@ LibraryManager.library = {
 #endif
     // void *dlopen(const char *file, int mode);
     // http://pubs.opengroup.org/onlinepubs/009695399/functions/dlopen.html
-    filename = filename === 0 ? '__self__' : (ENV['LD_LIBRARY_PATH'] || '/') + Pointer_stringify(filename);
+    var searchpaths = [];
+    if (filename === 0) {
+      filename = '__self__';
+    } else {
+      var strfilename = Pointer_stringify(filename);
+      if (strfilename.length > 0 && strfilename.substr(0,1) == '/') {
```

Comment:
Instead of special-casing an absolute path, I think you can use `PATH.join2(dir, strfilename)` to join them, and it will handle the case of strfilename being absolute for you.